src/main/java/net/earthcomputer/clientcommands/command/MapCommand.java

```diff
@@ -0,0 +1,269 @@
+package net.earthcomputer.clientcommands.command;
+
+import com.mojang.brigadier.Command;
+import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.exceptions.SimpleCommandExceptionType;
+import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource;
+import net.minecraft.block.MapColor;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.ClientPlayerEntity;
+import net.minecraft.client.texture.NativeImage;
+import net.minecraft.client.util.ScreenshotRecorder;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.decoration.ItemFrameEntity;
+import net.minecraft.item.FilledMapItem;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.map.MapState;
+import net.minecraft.text.ClickEvent;
+import net.minecraft.text.Text;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.util.math.Vec3i;
+import net.minecraft.world.World;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static com.mojang.brigadier.arguments.IntegerArgumentType.*;
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.*;
+
+public class MapCommand {
+    private static final SimpleCommandExceptionType NO_HELD_MAP_EXCEPTION = new SimpleCommandExceptionType(Text.translatable("commands.cmap.noHeld"));
+
+
+    public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher) {
+        dispatcher.register(literal("cmap")
+            .then(literal("export")
+                .executes(ctx -> exportMap(ctx.getSource(), 1))
+                .then(argument("upscale", integer(1, 16))
+                    .executes(ctx -> exportMap(ctx.getSource(), getInteger(ctx, "upscale")))
+                )
+            )
+        );
+    }
+
+    private static MapInfo[][] getMaps(ClientPlayerEntity player, MinecraftClient client) throws CommandSyntaxException {
+        MapInfo hand = fromHand(player);
+        if (hand != null) {
+            return new MapInfo[][]{{hand}};
+        }
+        MapInfo[][] mapInfo = fromWorld(player, client.targetedEntity);
+        if (mapInfo == null) {
+            throw NO_HELD_MAP_EXCEPTION.create();
+        }
+        return mapInfo;
+    }
+
+
+    private static int exportMap(FabricClientCommandSource source, int upscale) throws CommandSyntaxException {
+        MapInfo[][] mapInfo = getMaps(source.getPlayer(), source.getClient());
+        // calculate width and height
+        int width = mapInfo[0].length * 128 * upscale;
+        int height = mapInfo.length * 128 * upscale;
```

Comment:
Could `128` be replaced with size constants? These constants could be initialized to `FilledMapItem.field_30907` and `FilledMapItem.field_30908`.

Reply:
which one is which

Reply:
I'd assume `FilledMapItem.field_30907` is width and `FilledMapItem.field_30908` is height.